pilot/proxy/envoy/watcher.go

```diff
@@ -112,26 +116,28 @@ func (w *watcher) Reload() {
 
 // retrieveAZ will only run once and then exit because AZ won't change over a proxy's lifecycle
 // it has to use a reload due to limitations with envoy (az has to be passed in as a flag)
-func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration) {
-	for w.config.AvailabilityZone == "" {
+func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration, retries int) {
+	attempts := 0
+	for w.config.AvailabilityZone == "" && attempts <= retries {
 		time.Sleep(delay)
 		resp, err := http.Get(fmt.Sprintf("http://%v/v1/az/%v/%v", w.config.DiscoveryAddress, w.config.ServiceCluster, w.role.ServiceNode()))
```

Comment:
I assume the /az/ is an Istio-specific API, taking service node and returning some node-specific settings ? 

Is there a doc describing the API ? Can we extend it, to return more - like mixer/ca addresses to use for the node,
and other mesh or node-specific settings that go into the initial template ?

Reply:
interesting thought.. its istio specific

Reply:
Can be extended sure, but thats probably a discussion for another place! Feel free to open up an issue with your thoughts on the matter and we can discuss there.